src/mon/LogMonitor.cc

```diff
@@ -403,13 +403,44 @@ bool LogMonitor::preprocess_command(MonOpRequestRef op)
     if (f) {
       f->open_array_section("tail");
     }
+
+    std::string level_str;
+    clog_type level;
+    if (cmd_getval(g_ceph_context, cmdmap, "level", level_str)) {
+      level = LogEntry::str_to_level(level_str);
+      if (level == CLOG_UNKNOWN) {
+        ss << "Invalid severity '" << level_str << "'";
+        mon->reply_command(op, -EINVAL, ss.str(), get_last_committed());
+        return true;
+      }
+    } else {
+      level = CLOG_INFO;
+    }
+
+    std::string channel;
+    if (!cmd_getval(g_ceph_context, cmdmap, "channel", channel)) {
+      channel = CLOG_CHANNEL_DEFAULT;
+    }
+
+    // We'll apply this twice, once while counting out lines
+    // and once while outputting them.
+    auto match = [level, channel](const LogEntry &entry) {
+      return entry.prio >= level && (entry.channel == channel || channel == "*");
```

Comment:
this breaks the test of https://github.com/ceph/ceph/blob/master/qa/workunits/cephtool/test.sh#L709-L711. see http://pulpito.ceph.com/kchai-2017-06-28_07:28:49-rados-wip-kefu-testing-distro-basic-smithi/1337521/

probably, we should have
```diff
diff --git a/src/mon/LogMonitor.cc b/src/mon/LogMonitor.cc
index 319196b6c5..0d0fd63dbc 100644
--- a/src/mon/LogMonitor.cc
+++ b/src/mon/LogMonitor.cc
@@ -500,6 +500,7 @@ bool LogMonitor::prepare_command(MonOpRequestRef op)
     le.seq = 0;
     le.prio = CLOG_INFO;
     le.msg = str_join(logtext, " ");
+    le.channel = CLOG_CHANNEL_DEFAULT;
     pending_summary.add(le);
     pending_summary.prune(g_conf->mon_log_max_summary);
     pending_log.insert(pair<utime_t,LogEntry>(le.stamp, le));
```
also?